llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -6843,6 +6845,24 @@ SDValue RISCVTargetLowering::getDynamicTLSAddr(GlobalAddressSDNode *N,
   return LowerCallTo(CLI).first;
 }
 
+SDValue
+RISCVTargetLowering::getGeneralDynamicTLSDescAddr(GlobalAddressSDNode *N,
```

Comment:
`GeneralDynamicTLSDesc` in the name` `getGeneralDynamicTLSDescAddr` is misleading. general-dynamic refers to a traditional TLS mode while TLSDESC is entirely different.

Reply:
Well, AArch64 handles lowering for llvm's general-dynamic and local-dynamic cases differently. I was trying to make it clear here that we're not trying to do anything fancy like elide loads of the `__MODULE_BASE__` for multiple accesses.  I'm not convinced we want to do that in RISC-V, so it's probably fine to rename this.

Maybe `getTLSDescAddr` is good enough? at least until we decide that we want to optimize them differently.

Reply:
getTLSDescAddr sounds good.

Reply:
done.